tools/rccl-prim-test/rccl_prim_test.cpp

```diff
@@ -188,194 +276,164 @@ int main(int argc,char* argv[])
-      /*stream*/                stream_0,
-      /*kernel args*/           buff_coarse_0, 2*N, 0);
-
-  hipDev = 1;
-  HIPCHECK(hipSetDevice(hipDev));
-  HIPCHECK(hipGetDeviceProperties(&prop, hipDev));
-  printf("#   device %d [0x%02x] %s\n",
-                  hipDev, prop.pciBusID, prop.name);
-  HIPCHECK(hipExtMallocWithFlags((void**) &transfer_data_1, sizeof(struct transfer_data_t), hipDeviceMallocFinegrained));
-  //printf("GPU 1: allocated fine grain VRAM at %llx\n", (unsigned long long)transfer_data_1);
-  HIPCHECK(hipExtMallocWithFlags((void**) &buff_1, 2*N*sizeof(float), hipDeviceMallocFinegrained));
-  //printf("GPU 1: allocated fine grain VRAM at %llx\n", (unsigned long long)buff_1);
-  HIPCHECK(hipMalloc((void**) &buff_coarse_1, 2*N*sizeof(float)));
-  //printf("GPU 1: allocated coarse grain VRAM at %llx\n", (unsigned long long)buff_coarse_1);
-  profiling_data_1 = (struct profiling_data_t *)malloc(sizeof(struct profiling_data_t));
-  HIPCHECK(hipMalloc((void**) &d_profiling_data_1, sizeof(struct profiling_data_t)));
-  //create stream
-  hipStream_t stream_1;
-  HIPCHECK(hipStreamCreate(&stream_1));
-  //randomize test data
-  hipLaunchKernelGGL(initTestDataKernel,
-      /*grid dim x,y,z*/        dim3(32, 1, 1),
-      /*block dim x,y,z*/       dim3(THREADS, 1, 1),
-      /*dynamic shared mem*/    0,
-      /*stream*/                stream_1,
-      /*kernel args*/           buff_1, 2*N, 1);
-  hipLaunchKernelGGL(initTestDataKernel,
-      /*grid dim x,y,z*/        dim3(32, 1, 1),
-      /*block dim x,y,z*/       dim3(THREADS, 1, 1),
-      /*dynamic shared mem*/    0,
-      /*stream*/                stream_1,
-      /*kernel args*/           buff_coarse_1, 2*N, 1);
-
-  h_transfer_data_0.dest0 = buff_1;
-  h_transfer_data_0.dest1 = buff_coarse_0 + N;
-  h_transfer_data_0.src0 = buff_0;
-  h_transfer_data_0.src1 = buff_coarse_0;
-  h_transfer_data_0.N = N;
-  h_transfer_data_0.gpu = 0;
-
-  h_transfer_data_1.dest0 = buff_0 + N;
-  h_transfer_data_1.dest1 = buff_coarse_1;
-  h_transfer_data_1.src0 = buff_1 + N;
-  h_transfer_data_1.src1 = buff_coarse_1 + N;
-  h_transfer_data_1.N = N;
-  h_transfer_data_1.gpu = 1;
-
-  HIPCHECK(hipSetDevice(0));
-  HIPCHECK(hipMemcpyAsync(transfer_data_0, &h_transfer_data_0,
-                          sizeof(struct transfer_data_t), hipMemcpyHostToDevice,
-                          stream_0));
-  HIPCHECK(hipStreamSynchronize(stream_0));
-
-  HIPCHECK(hipSetDevice(1));
-  HIPCHECK(hipMemcpyAsync(transfer_data_1, &h_transfer_data_1,
-                          sizeof(struct transfer_data_t), hipMemcpyHostToDevice,
-                          stream_1));
-  HIPCHECK(hipStreamSynchronize(stream_1));
+  float *buff[MAX_GPU*MAX_WORKGROUPS], *buff_coarse[MAX_GPU*MAX_WORKGROUPS];
+  struct transfer_data_t h_transfer_data[MAX_GPU], *transfer_data[MAX_GPU];
+  struct profiling_data_t *profiling_data[MAX_GPU], *d_profiling_data[MAX_GPU];
+  hipStream_t stream[MAX_GPU];
+
+  uint64_t N = 2097152;
```

Comment:
Let's make this accessible via a cmd line parameter as well.

Reply:
done